packages/council-graphql/src/resolvers/index.ts

```diff
@@ -59,40 +64,67 @@ export const resolvers: Resolvers<CouncilContext> = {
         voters = await VoterModel.getAll({ context });
       }
 
-      if (search) {
-        const candidates: string[] = [];
-
-        voters = await Promise.all(
-          voters.map(async (voter) => {
-            if (!voter) {
-              return;
-            }
+      if (search || votingPowerMin || votingPowerMax) {
+        let vaults: (VotingVault | undefined)[];
 
-            candidates.push(voter.address);
+        for (let voter of voters) {
+          if (!voter) {
+            continue;
+          }
 
+          // Get ensName of voter
+          if (search) {
             const ensName = await VoterModel.getEnsName({ voter, context });
-            if (ensName) {
-              candidates.push(ensName);
+            voter = { ...voter, ensName };
+          }
+
+          // Get votingPower of voter
+          if (votingPowerMin || votingPowerMax) {
+            if (votingVaults) {
+              vaults = VotingVaultModel.getByAddresses({
+                addresses: votingVaults,
+                context,
+              }).filter((vault) => vault !== undefined);
+            } else {
+              vaults = VotingVaultModel.getAll({ context });
```

Comment:
If we get the vaults earlier (see previous comment), then here we only need to get all if they weren't already fetched.